addons/mail/static/src/components/thread_view/thread_view.xml

```diff
@@ -23,6 +23,13 @@
                         t-ref="messageList"
                     />
                 </t>
+                <t t-elif="threadView.threadCache.hasLoadingFailed">
+                    <div class="o_ThreadView_error">
+                        <span class="o_ThreadView_errorText">
```

Comment:
Use better name for part. `error` is quite generic... `loadingFailed` is better.
Side-note that we plan moving putting more concerns in thread cache so that it's not just about messages (but also attachments and activities), so it's not super clear it's about failed to load messages but it's fine for now.